### PR TITLE
refactor: migrate `configtest` package to ESM

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
refactor.

**Did you add tests for your changes?**
No
**If relevant, did you update the documentation?**
No
**Summary**
Start ESM migration.

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

**Review by evenstensberg — COMMENTED** [review]
Maybe draft these ESM converted packages and group them up, until every package is done migrated?

**Comment by snitin315** [comment]
Yes, I'll convert this to a draft.

**Comment by alexander-akait** [comment]
@snitin315 I think in this PR we need to convert all to ESM (or alternative we can use `import(...)`) to load extensions for webpack-cli, so they will work

File: packages/configtest/package.json [hunk 1]
```diff
@@ -3,7 +3,12 @@
   "version": "1.2.0",
   "description": "Validate a webpack configuration.",
   "type": "module",
-  "exports": "./lib/index.js",
+  "exports": {
+    "./package.json": "./package.json",
+    ".": {
+      "import": "./dist/index.js"
```
Comment (alexander-akait) [inline packages/configtest/package.json:9]: Always add `default` as fallback :+1:

Comment (snitin315) [reply packages/configtest/package.json:9]: Yes, webpack v5 is using [require](https://github.com/webpack/webpack/blob/9fcaa243573005d6fdece9a3f8d89a0e8b399613/bin/webpack.js#L69) as well.

**Comment by snitin315** [comment]
@alexander-akait I suppose if we move to ESM it will break compatibility with webpack v5 too. Because [webpack 5 is using cli with `require`](https://github.com/webpack/webpack/blob/9fcaa243573005d6fdece9a3f8d89a0e8b399613/bin/webpack.js#L69)

Screenshot 2022-11-06 at 4 05 28 PM

**Comment by alexander-akait** [comment]
What about if we will use `import(...)`? It should work

**Comment by snitin315** [comment]
@alexander-akait yes but it means from webpack v5.0.0 to the current latest version of webpack will not work with this major version. It will be compatible only with future version of webpack which is using import().

**Comment by snitin315** [comment]
@alexander-akait Yes `import(...)` will work. But it means users will have to update webpack to the latest version which includes this PR https://github.com/webpack/webpack/pull/16448

**Comment by alexander-akait** [comment]
@snitin315 Oh, I got it, yeah, let's postpone it then, and mark it for webpack v6, we will need to do it for webpack-dev-server too

**Comment by snitin315** [comment]
@alexander-akait yes, let's postpone.

**Comment by alexander-akait** [comment]
Let's close in favor https://github.com/webpack/webpack-cli/issues/3436, I think we will start migration everywhere on ESM when Node.js@18 will be outdated, because there are some bugs in Node.js@18 with ESM and especial with `import(...)`, we can make a pain for many developers if we do it right now, anyway thank you

**Pull request closed without merging**